### DISCUSSION TITLE
How to make useFetch body reactive using GraphQL?

### DISCUSSION THREAD

**Discussion by arhubi** [nuxt/nuxt#17367] [Questions] [answered]
Hi,
I'm developing a "search results" page in Nuxt 3 and I'm trying from it to do a re-fetch using the useFetch composable.

This is how I use the useFetch function:
```typescript
const queryBody = computed(() => ({
  query: searchQuery.loc.source.body, // imported query from a .graphql file
  variables: {
    name: searchText.value,
  },
}));

const { data, refresh } = await useFetch(config.API_URL, {
  method: 'POST',
  headers: {
    Authorization: `Bearer ${config.API_TOKEN}`,
  },
  body: queryBody.value,
});

// ...later
const search = () => {
  refresh();
};
```

The issue I'm facing is when I change the queryBody.value, the body of the request is logically not changed when I trigger `refresh` as I pass the queryBody as value. Is there a way to pass it as ref and make the useFetch reactive? 
If not, what is the best way to achieve what I want to do using composables? Do I have to write a custom one?

Thank you,

**Comment by benneq** [comment, marked as answer]
I came across a similar issue. You can workaround this using `useAsyncData` and `$fetch`:

```
const { pending, data, refresh } = useAsyncData('needToProvideAKey', () => {
    return $fetch(config.API_URL, { body: queryBody.value });
});
```

This way the whole `$fetch` code will be re-run everytime you call `refresh`.

**Reply by wobsoriano** [reply]
This is the correct answer. As per [useFetch](https://v3.nuxtjs.org/docs/usage/data-fetching/#usefetch) docs - `It automatically generates a key based on URL and fetch options, as well as infers API response type.` 

Looking at his code, the URL and options aren't updated that's why the result is cached.

**Reply by AaronConlon** [reply]
You are right.Thanks!
